src/components/taskList/TaskList.js

```diff
@@ -0,0 +1,82 @@
+import './taskList.scss';
+
+class TaskList extends React.PureComponent {
+  state = {
+    value: '',
+    todo: [],
+    originTodos: []
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.content !== prevState.originTodos) {
+      return { originTodos: nextProps.content, todo: nextProps.content, value: '' };
+    }
+  }
+
+  changeHandler = (event) => {
+    const { originTodos } = this.state;
+    this.setState({
+      todo: originTodos.filter((el) => {
+        if (event.target.value.length > 1) {
+          return el.title.toLowerCase().includes(event.target.value.toLowerCase());
+        }
+        return { todo: originTodos };
+      }),
+      value: event.target.value
+    });
+  }
+
+  taskControlHandler = (event) => {
+    console.log(event.target.className);
+  }
+
+  render() {
+    const {
+      todo,
+      value
+    } = this.state;
+
+    const { markDone, delTask } = this.props;
+
+    const output = (el) => {
+      return (
+        <li
+          key={el.id}
+          className={!el.done ? 'uncompleted' : 'completed'}
+          onMouseEnter={this.showControls}
+        >
+          <span>
+            <a href="#">{el.title}</a>
+            {
+              !el.done
+              && <>
+                <span className="elemControl delete" onClick={() => delTask(el.id)} />
+                <span className="elemControl done" onClick={() => { el.done = true; return markDone(el); }} />
```

Comment:
it is not good practice to place function body directly to jsx `onClick={() => { el.done = true; return markDone(el); }}`